Samples/BasicTelemetry.sample.json

```diff
@@ -19,7 +19,7 @@
                     "default_when_enabled": "True_Override"
                 },
                 "telemetry": {
-                    "enabled": "true",
+                    "enabled": true,
```

Comment:
Would this break any test?
If not, maybe we should add a test to validate this.

Reply:
Maybe a better question for @rossgrambo or @zhiyuanliang-ms. I've updated all of them to be Boolean  in this PR, but Python and Spring don't care if they are Strings or Booleans and just make them work.